src/coordinator/ts/values.go

```diff
@@ -133,3 +139,39 @@ func NewFixedStepValues(millisPerStep time.Duration, numSteps int, initialValue
 		values:        values,
 	}
 }
+
+// At every resolution timestep, the last sample value is chosen
+func RawPointsToFixedStep(datapoints Datapoints, start time.Time, end time.Time, interval time.Duration) (FixedResolutionMutableValues, error) {
+	if end.Before(start) {
+		return nil, fmt.Errorf("start cannot be after end, start: %v, end: %v", start, end)
+	}
+	numSteps := int(end.Sub(start) / interval)
+	if end == start {
+		numSteps = 1
+	}
+	fixStepValues := newFixedStepValues(interval, numSteps, math.NaN(), start)
+	currIdx := 0
+	idx := 0
+	for t := start; !t.After(end) && currIdx < numSteps; t = t.Add(interval) {
+		// Find first datapoint not before time t
+		for ; idx < len(datapoints) && datapoints.DatapointAt(idx).Timestamp.Before(t); idx++ {
+		}
+
+		if idx >= len(datapoints) {
```

Comment:
nit: idx can never be greater than len(datapoints)

Reply:
based on L157, idx can be = len(datapoints). Since we maintain idx across multiple for loops, i'm just using it as sanity check